From 1f13f721263a4cc75e4b798a230022561047899c Mon Sep 17 00:00:00 2001
From: Eduard Bagdasaryan <eduard.bagdasaryan@measurement-factory.com>
Date: Thu, 1 Dec 2022 18:50:37 +0000
Subject: [PATCH] Bug 5162: mgr:index URL do not produce MGR_INDEX template
 (#1191)

Satisfy mgr:index requests using

* a 200 OK response with a body derived from the MGR_INDEX template (if
  that template file was found during (re)configuration) or
* a 404 (Not Found) error response (otherwise).

Broken in 2019 commit 7e6eabb, when Squid started replying using a 200
OK response with a hard-coded "mgr_index" text as a body, ignoring any
configured MGR_INDEX template.
---
 src/errorpage.cc | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/errorpage.cc b/src/errorpage.cc
index 6fbedbe1dba..f74e6e554e2 100644
--- a/src/errorpage.cc
+++ b/src/errorpage.cc
@@ -154,6 +154,7 @@ static const struct {
     const char *text;
 }
 
+/// error messages that cannot be configured/customized externally
 error_hard_text[] = {
 
     {
@@ -180,10 +181,6 @@ error_hard_text[] = {
     {
         ERR_REQUEST_START_TIMEOUT,
         "request start timedout"
-    },
-    {
-        MGR_INDEX,
-        "mgr_index"
     }
 };
 
